sycl/unittests/kernel-and-program/DeviceInfo.cpp

```diff
@@ -0,0 +1,80 @@
+//==-------------- DeviceInfo.cpp --- device info unit test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#define SYCL2020_DISABLE_DEPRECATION_WARNINGS
+
+#include <CL/sycl.hpp>
+#include <detail/context_impl.hpp>
+#include <gtest/gtest.h>
+#include <helpers/PiMock.hpp>
+
+using namespace sycl;
+
+namespace {
+struct TestCtx {
+  TestCtx(context &Ctx) : Ctx{Ctx} {};
+
+  context &Ctx;
+};
+} // namespace
+
+static std::unique_ptr<TestCtx> TestContext;
+
+static pi_result redefinedDeviceGetInfo(pi_device device,
+                                        pi_device_info param_name,
+                                        size_t param_value_size,
+                                        void *param_value,
+                                        size_t *param_value_size_ret) {
+  return PI_SUCCESS;
```

Comment:
Could you please return some result for `param_name` == `PI_DEVICE_INFO_UUID` and check this result on line 76?

Reply:
done.

Reply:
Sorry if I was not clear. I meant that the function could return some UUID, e.g. 0xFF, and this value can be checked in the main function. I'm OK if this improvement is done in a separate PR.

Reply:
the change did not got uploaded, my bad. I am uploading it now.

Reply:
@romanovvlad please review now.

Reply:
upon discussion with vlad, we've decided to have the change on a diffPR.